pkg/dao/db.go

```diff
@@ -1,20 +1,55 @@
 package dao
 
-import "database/sql"
+import (
+	"database/sql"
+	"errors"
+	"fmt"
 
-var db *sql.DB
+	_ "github.com/lib/pq"
+)
 
-func Connect() (*sql.DB, error) {
-	if db != nil {
-		return db, nil
+type DatabaseHandler struct {
+	DB *sql.DB
+}
+
+func (handler *DatabaseHandler) Connect(dbHost, dbUser, dbPassword, dbName, sslMode string) error {
+	if handler.DB != nil {
+		return nil
 	}
 	//connection logic
-	return nil, nil
+	connStr := fmt.Sprintf("host=%v user=%v password=%v dbname=%v sslmode=%v", dbHost, dbUser, dbPassword, dbName, sslMode)
```

Comment:
may want to consider looking into setting values for 

```
db.SetMaxOpenConns()
db.SetConnMaxLifetime()
```
as otherwise you can end up with invalid connections. Have a read of what they do and figure out what makes sense for you

Reply:
+1 I was reading up on them yesterday but I totally forgot about it today!